### PR TITLE
First draft of cKDTree KNN algorithm / tech improvement section

### PR DESCRIPTION
**Pull request by tylerjereddy**
It would be helpful to carefully check the literature citations (probably add more / correct ones) & perhaps add in the worst-case time complexity for the KNN algorithm. I can find a few informal derivations of this online, but I'm not convinced that I've found a published manuscript that corresponds to what was implemented in SciPy for KNN.

### CODE REVIEW

File: scipy-1.0/ckdtree.tex [hunk 1]
```diff
@@ -5,8 +5,21 @@
 
 \subsubsection{k-NN}
 
-The API for querying neighbours is enhanced, to allow querying a subset of
-k nearest neighbours.
+% TODO: please check / improve literature citations used here
+% what about the time complexity of the kNN implementation?
+In 2015, we enhanced \texttt{cKDTree.query} with a $k$ nearest neighbors search
+parameter. This may be an efficient operation\cite{Sproull:1991:RNS:3118219.3118331}
```
Comment (andyfaff) [inline scipy-1.0/ckdtree.tex]: may be --> is

File: scipy-1.0/ckdtree.tex [hunk 2]
```diff
@@ -5,8 +5,21 @@
...
+because it generates and returns a data structure that may be only a fraction 
+of the size of the full neighbor list. A C++ implementation is provided that releases
+the Python global interpreter lock for the neighbor search process, uses
+a memory pool to allocate and automatically reclaim \texttt{structs}, and
+heapsort to sort the $k$ nearest neighbors distances, even in the case
+of periodic boundary conditions. When $k$ is a non-contiguous list of nearest
+neighbors to account for (i.e., \texttt{[1, 3, 4]}), the intervening distances
+between the nearest neighbor ($k = 1$) and the farthest neighbor requested
+are all interrogated, but only those neighbors that are requested are retained
+in memory. Thus, the max heap size for this algorithm is conceptually determined
```
Comment (andyfaff) [inline scipy-1.0/ckdtree.tex]: max --> maximum

**Comment by tylerjereddy** [comment]
Thanks @andyfaff ; revised accordingly

**Comment by rgommers** [comment]
Looking at gh-38 and gh-39, these are both about cKDTree and probably should be merged or we should drop one. Hard to do via PR review though, so I'll merge both and then we can edit the paper draft afterwards.

**Pull request merged**